app/views/NewServerView/ServerInput/index.tsx

```diff
@@ -72,14 +74,15 @@ const ServerInput = ({
 						{ backgroundColor: themes[theme].backgroundColor, borderColor: themes[theme].separatorColor }
 					]}
 				>
-					<FlatList
-						data={serversHistory}
-						renderItem={({ item }) => (
+					{/* NB: in theory the .slice(0, MAX_ITEMS_HISTORY) is useless because the number of items in the history is already limited to three when querying the database: */}
+					{/*  https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/5aaf7af5f4004e8b0b9386da579ae4c8aed3a9a0/app/views/NewServerView/index.tsx#L160 */}
+					{/* But we repeat the filtering here, since having more than three items would break the display pretty badly. */}
+					{serversHistory.slice(0, MAX_ITEMS_HISTORY).map(item => (
+						<View key={item.id}>
 							<Item item={item} theme={theme} onPress={() => onPressServerHistory(item)} onDelete={onDelete} />
-						)}
-						ItemSeparatorComponent={List.Separator}
-						keyExtractor={item => item.id}
-					/>
+							<Separator />
```

Comment:
This way you're doubling border bottom
![image](https://user-images.githubusercontent.com/804994/225967638-1433964a-ed55-45fe-8725-e1ea5c685bcc.png)